### DISCUSSION TITLE
Rebase from target branch results in unsigned commits

### DISCUSSION THREAD

**Discussion by jonathanmorley** [community/community#64125] [Pull Requests]
### Select Topic Area

Bug

### Body

I have 'vigilant mode' enabled, and I had a PR where I had 19 signed commits pushed.
The target branch (main) was updated, and GitHub gave me the opportunity from within the PR to update the PR to reflect those changes.

I had the option of creating a merge commit, or rebasing.

Previously, I had chosen a merge commit, and the merge commit that was created was signed.
This time, I chose rebase, and the 19 new commits (one per commit in the PR) that it created were all unsigned.

This was unexpected.

**Comment by vacisdev** [comment]
During the rebase operation, commit signatures can be lost as new commits are created. To ensure that the new commits are signed, you may need to manually sign them or enable automatic signing for rebased commits in your Git configuration.

**Reply by jonathanmorley** [reply]
The value of the button in the PR UI to auto-rebase the PR seems lower if I will then have to manually sign those commits. Especially as the merge commit looks to be auto signed by github, it seems like a bug that the rebased commits are not

**Comment by jordanst3wart** [comment]
duplicates https://github.com/orgs/community/discussions/11639
